### PR TITLE
Add dns-packet module for code review

### PR DESCRIPTION
**Pull request by srsudar**

### CODE REVIEW

File: chromeapp/app/scripts/dnssd/dns-packet.js [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+ * Illustrated, Volume 1: The Protocols' by Stevens.
+ */
+'use strict';
+
+var resRec = require('./resource-record');
+var dnsCodes = require('./dns-codes');
+var byteArray = require('./byte-array');
+var qSection = require('./question-section');
+
+var MAX_ID = 65535;
```
Comment (mdwelsh) [inline chromeapp/app/scripts/dnssd/dns-packet.js:24]: Consider adding comments for these constants and what they mean/where they come from.

File: chromeapp/test/scripts/dnssd/dns-packet-test.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
+'use strict';
```
Comment (mdwelsh) [inline chromeapp/test/scripts/dnssd/dns-packet-test.js:1]: You need unit tests for parseResourceRecordsFromReader, covering the error cases too.

File: chromeapp/test/scripts/dnssd/dns-packet-test.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+var test = require('tape');
+var dnsPacket = require('../../../app/scripts/dnssd/dns-packet');
+var resRec = require('../../../app/scripts/dnssd/resource-record');
+var qSection = require('../../../app/scripts/dnssd/question-section');
+
+test('can create a DnsPacket', function(t) {
```
Comment (mdwelsh) [inline chromeapp/test/scripts/dnssd/dns-packet-test.js]: You should probably have tests covering various expected error cases as well.

File: chromeapp/app/scripts/dnssd/dns-packet.js (hunk 1 above)
Comment (srsudar) [inline chromeapp/app/scripts/dnssd/dns-packet.js:24]: done

File: chromeapp/test/scripts/dnssd/dns-packet-test.js (hunk 3 above)
Comment (srsudar) [inline chromeapp/test/scripts/dnssd/dns-packet-test.js]: done

File: chromeapp/test/scripts/dnssd/dns-packet-test.js (hunk 2 above)
Comment (srsudar) [inline chromeapp/test/scripts/dnssd/dns-packet-test.js:1]: Exposed it on the public API (I'd kept it internal for some reason) and added tests.

done

**Pull request merged**